app/src/main/java/io/homeassistant/companion/android/onboarding/integration/MobileAppIntegrationFragment.kt

```diff
@@ -87,8 +96,32 @@ class MobileAppIntegrationFragment : Fragment(), MobileAppIntegrationView {
 
         if (PermissionManager.validateLocationPermissions(requestCode, permissions, grantResults)) {
             presenter.onGrantedLocationPermission(context!!, activity!!)
+            requestBackgroundAccess()
+        } else {
+            (activity as MobileAppIntegrationListener).onIntegrationRegistrationComplete()
         }
+    }
 
-        (activity as MobileAppIntegrationListener).onIntegrationRegistrationComplete()
+    @SuppressLint("BatteryLife")
+    private fun requestBackgroundAccess() {
+        val packageName = activity?.packageName ?: ""
+        val intent: Intent
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M &&
+            context?.getSystemService(PowerManager::class.java)?.isIgnoringBatteryOptimizations(packageName) == false
+        ) {
+            intent = Intent(
```

Comment:
Can we inform the user why we're requesting this ? I wonder if we need to create some checklist page with buttons the user can press to be informed about it:

![image](https://user-images.githubusercontent.com/1444314/72672293-a88c3180-3a0c-11ea-88eb-4db4bae70a6a.png)